src/templates/users/user_detail.html

```diff
@@ -3,25 +3,43 @@
 {% load static %}
 
 {% block title %}
-  User:
-  {{ object.name }}
+  My Details:
 {% endblock title %}
 {% block content %}
   <div class="container">
     <div class="row">
       <div class="col-sm-12">
-        <h2>{{ object.name }}</h2>
+        <h2>My Details:</h2>
       </div>
     </div>
     {% if object == request.user %}
```

Comment:
currently a user can use the url and get to the page but they won't see personal info which is great but if you want to prevent users from accessing others personal info pages, you can use  UserPassesTestMixin in the class-based view where the test_func would be something like

`
def test_func(self):
        return self.request.user == self.get_object()
`

https://docs.djangoproject.com/en/4.2/topics/auth/default/#django.contrib.auth.mixins.UserPassesTestMixin

also if you want to control what happens to "illegal" users, you can use AccessMixin and change the get_permission_denied_message()

https://docs.djangoproject.com/en/4.2/topics/auth/default/#django.contrib.auth.mixins.AccessMixin

Reply:
Thank you for the idea, I just implemented it, and it works as expected.

Reply:
Hey, so after pushing my code, I'm failing one test in which the test case TestUserDetailView.test_authenticated is failing due to a PermissionDenied exception being raised. This happens when the UserPassesTestMixin in my UserDetailView denies permission for the test user, which it's doing properly, but since the test is formatted this way, I can't implement this properly.

Reply:
Which test is it failing? If the test is looking for access to a page that it should not have access to, maybe we should actually change the test.